### PR TITLE
[DYN-7839] Wires & Pins: improve snapping tolerance

### PR DESCRIPTION
**Pull request by ivaylo-matov**
### Purpose

PR aims to address [DYN-7839](https://jira.autodesk.com/browse/DYN-7839).

This update enhances the snapping tolerance for connectors when interacting with the ConnectorAnchor.

A second, invisible connector has been introduced to handle MouseEnter and MouseLeave events. This invisible connector features a dynamic `StrokeThickness` that adjusts based on the current zoom level:

Zoom > 1: The `StrokeThickness` matches the visible connector's thickness (default: 3).
Zoom < 1: The `StrokeThickness` dynamically scales with the zoom factor, ensuring that the actual snapping area on the screen remains consistent regardless of the zoom level.
The `StrokeThickness` is doubled at smaller zoom factors to improve the responsiveness.

DYN-7839-Fix

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

Improved connector snapping at lower zoom levels for Pin and Watch buttons.

### Reviewers
@QilongTang 
@reddyashish 

### FYIs
@dnenov

### CODE REVIEW

**Comment by ivaylo-matov** [comment]
Wasn't sure

**Comment by QilongTang** [comment]
?

**Comment by QilongTang** [comment]
hi @ivaylo-matov Would love more time reviewing this PR, curious about the performance impact of the second connector here.

**Comment by ivaylo-matov** [comment]
@achintyabhat, I’ve implemented scaling functionality for the anchor, pin, and watch buttons. The size of these buttons now decreases as you zoom out with a non-linear scaling behavior. At Zoom = 1, the buttons retain their default size, but at smaller scales, the size gradually reduces to half of their original dimensions. IMO keeping the default button size at smaller scales looks odd. With this scaling behavior, the buttons remain comfortably clickable, even at zoom levels where the connector is no longer visible.

DYN-7839-ButtonsFix


@QilongTang, I compared the opening times for Performance-Test.dyn against `master` and saw no degradation in performance.

File: src/DynamoCoreWpf/ViewModels/Preview/ConnectorAnchorViewModel.cs [hunk 1]
```diff
@@ -29,16 +30,34 @@ public class ConnectorAnchorViewModel: ViewModelBase
...
+        /// </summary>
+        public double AnchorSize => GetCachedScaledSize(15);
+        /// <summary>
+        /// The offset for the anchor button, calculated as half the size.
+        /// </summary>
+        public double AnchorOffset => AnchorSize / -2;
+        /// <summary>
+        /// The vertical offset for the Watch and Pin buttons.
         /// </summary>
-        public double MarkerSize { get; set; } = 30;
```
Comment (reddyashish) [inline src/DynamoCoreWpf/ViewModels/Preview/ConnectorAnchorViewModel.cs:40]: API break due to removing the setter property.

**Review by reddyashish — APPROVED** [review]

**Pull request merged**